internal/pkg/describe/lb_web_service.go

```diff
@@ -289,36 +300,79 @@ type WebServiceRoute struct {
 	URL         string `json:"url"`
 }
 
-// ServiceDiscovery contains serialized service discovery info for an service.
+// ServiceDiscovery contains serialized service discovery info for an ECS service.
 type ServiceDiscovery struct {
 	Environment []string `json:"environment"`
-	Namespace   string   `json:"namespace"`
+	// Namespace is not an accurate name but unfortunately we can't take it back.
+	Namespace string `json:"namespace"`
```

Comment:
I am honestly so confused on what we are doing here 😝 
1. Does this mean this "namespace" is applicable for  all the `.Environments`?

I guess this structure exists because of our old service discovery setup with `{app}.local` 😭 .

I feel like our life would be so much simpler if we have:
```go
// A generic endpoint data type that can be used for both SC and SD.
type InternalEndpoint struct {
   Endpoint string 
   Environment string
}

// sdMarshaler is a collection of Service Discovery endpoints.
type sdMarshaler []*InternalEndpoint

// scMarshaler is a collection of Service Connect endpoints.
type scMarshaler []*InternalEndpoint


// MarshalJSON writes a collection of service discovery endpoints in JSON format.
func (sdMarshaler) MarshalJSON() ([]byte, error) {
    // for backwards compatibility define the old type here.
    type out struct {
        Namespace string
        Environments []string
    }
    //...
}
func (scMarshlaer) MarshalJSON() ([]byte, error)

// WriteHumanString writes a collection of service discovery endpoints grouped by environments.
func (sdMarshaler) WriteHumanString(w io.Writer) error { ... }
func (scMarshaler) WriteHumanString(w io.Writer) error { ... }

// collectEndpointsFor returns the collection of endpoints for a given environment.
func (sdMarshaler) collectEndpointsFor(env string) []string { ... }
```

Reply:
Namespace is wrong...it is a mistake we made before and we can't take it back because otherwise the json output will change. Namespace ideally should be only `env.app.local` whereas here the namespace is actually the full endpoint name

Reply:
TBH I'd rather switch to
```golang
type InternalEndpoint struct {
   Endpoint string 
   Environment string
}
```
for both sc and sd. change `Namespace` to `Endpoint` which makes more sense and we can argue bug fixing is not breaking change.